src/manifold/src/manifold.cpp

```diff
@@ -775,4 +778,61 @@ Manifold Manifold::TrimByPlane(glm::vec3 normal, float originOffset) const {
 }
 
 ExecutionParams& ManifoldParams() { return params; }
-}  // namespace manifold
\ No newline at end of file
+
+Manifold Manifold::Hull(const std::vector<glm::vec3>& pts,
+                        const std::vector<float>& props, const int numProp) {
+  const int numVert = pts.size();
+  // FIXME: Should there be a hull specific error?
+  if (numVert < 4) return Invalid();
+
+  std::vector<quickhull::Vector3<double>> vertices(numVert);
+  for (int i = 0; i < numVert; i++) {
+    vertices[i] = {pts[i].x, pts[i].y, pts[i].z};
+  }
+
+  quickhull::QuickHull<double> qh;
+  // bools: correct triangle winding, and use original indices
+  auto hull = qh.getConvexHull(vertices, false, true);
+  const auto& triangles = hull.getIndexBuffer();
+  const int numTris = triangles.size() / 3;
+
+  Mesh mesh;
+  Impl::MeshRelationD relation;
+  relation.numProp = numProp;
+  relation.properties = props;
+  mesh.vertPos = pts;
+
+  mesh.triVerts.reserve(numTris);
+  for (int i = 0; i < numTris; i++) {
+    const int j = i * 3;
+    mesh.triVerts.push_back({triangles[j], triangles[j + 1], triangles[j + 2]});
+  }
+  return Manifold(mesh);
+}
+
+Manifold Manifold::Hull() const {
+  auto mesh = GetMeshGL();
+  const int extraProps = mesh.numProp - 3;
+  const int numVert = mesh.NumVert();
+  std::vector<glm::vec3> pts(numVert);
+  std::vector<float> props;
+  if (extraProps > 0) props.reserve(extraProps * numVert);
+  for (int i = 0; i < mesh.NumVert(); i++) {
+    const int j = i * 3;
+    pts[i] = {mesh.vertProperties[j], mesh.vertProperties[j + 1],
+              mesh.vertProperties[j + 2]};
+    for (int k = 0; k < extraProps; k++) {
+      props.push_back(mesh.vertProperties[j + 3 + k]);
+    }
+  }
+  return Hull(pts, props, extraProps);
+}
+
+Manifold Manifold::Hull(const std::vector<Manifold>& manifolds) {
+  return BatchBoolean(manifolds, OpType::Add).Hull();
```

Comment:
I think a `Compose` would be a quite a bit faster here and I don't think you care about intersections.

Reply:
Right, I didn't think to use Compose, that makes more sense.